File: internal/store/overrides.go
package store

import "fmt"

// VerdictOverride is one durable record of a hand-set verdict from the UI: the
// delta (from → to), the reason the user gave, and the criteria version in
// effect at the time. Append-only; never updated, never written to the brain.
type VerdictOverride struct {
	CompanyID       string
	FromVerdict     string // prior verdict being replaced; "" if the company was unscored
	ToVerdict       string
	Reason          string
	CriteriaVersion string
}

// InsertVerdictOverride appends one override record. Unlike the verdict_trace
// rows (a disposable debug aid), this table is a record of intent.
func (db *DB) InsertVerdictOverride(o VerdictOverride) error {
	const q = `
INSERT INTO verdict_override (company_id, from_verdict, to_verdict, reason, criteria_version)
VALUES (?, ?, ?, ?, ?)`
	if _, err := db.Exec(q,
		o.CompanyID, nullIfEmpty(o.FromVerdict), o.ToVerdict, o.Reason, nullIfEmpty(o.CriteriaVersion),
	); err != nil {
		return fmt.Errorf("insert verdict override %s: %w", o.CompanyID, err)
	}
	return nil
}

File: scout/store/overrides.py
"""Durable hand-set verdict overrides (record of intent)."""

from __future__ import annotations

import sqlite3
from dataclasses import dataclass

from ._helpers import null


@dataclass
class VerdictOverride:
    company_id: str = ""
    from_verdict: str = ""  # prior verdict being replaced; "" if unscored
    to_verdict: str = ""
    reason: str = ""
    criteria_version: str = ""


def insert_verdict_override(con: sqlite3.Connection, o: VerdictOverride) -> None:
    """Append one override record."""
    con.execute(
        "INSERT INTO verdict_override (company_id, from_verdict, to_verdict, reason, criteria_version) "
        "VALUES (?, ?, ?, ?, ?)",
        (o.company_id, null(o.from_verdict), o.to_verdict, o.reason, null(o.criteria_version)),
    )
